pkg/controllers/user/controller.go

```diff
@@ -325,6 +336,27 @@ func (h *handler) removeKubeconfig(user *klum.User) error {
 	return nil
 }
 
+func (h *handler) OnKubeconfigChange(s string, kubeconfig *klum.Kubeconfig) (*klum.Kubeconfig, error) {
+	if kubeconfig == nil {
+		return nil, nil
+	}
+	user, err := h.kuser.Get(s, metav1.GetOptions{})
+	if err != nil {
+		log.Error(err)
+		return kubeconfig, nil
+	}
+
+	if h.cfg.GithubToken != "" {
+		err := github.UploadGithubSecret(kubeconfig, user, h.cfg.GithubURL, h.cfg.GithubToken)
+		if err != nil {
+			// Returning as if everything went well because we don't want to interfere with the kubeconfig creation
+			log.Error(err)
+			return kubeconfig, nil
+		}
+	}
+	return kubeconfig, nil
```

Comment:
Would be great to have the sync status in the CRD's `.status` field. Like "up-to-date" or "failed to sync"

Reply:
Introduced here [f83ac5c](https://github.com/jadolg/klum/pull/19/commits/f83ac5c06db2a442f69b529f850b8d38d4fb1d6a)

Reply:
I'm missing a lot of things in this framework.
The status thing is something we can definitely polish more in the future.